### PR TITLE
usage by time period metric

### LINKED ISSUE #230: Create API endpoint for "Usage by period" queries
See #215 - add an API endpoint for the "Usage by period" queries

### PR DESCRIPTION
**Pull request by peetucket**
## Why was this change made?

Fixes #230 - add api endpoints for various usage by time period (optionally filtered by group)

## How was this change tested?

Added new tests, localhost browser

### CODE REVIEW

File: src/endpoints/metrics.js [hunk 1]
```diff
@@ -22,8 +22,8 @@ const getResourceQuery = (resourceType) => {
       },
     }
   }
-  // This is reached by sending a value of "all" which is contrained by openapi
-  return null
+  // This is reached by sending "all" (we cannot get any other values since resourceType is an enum contrained by openapi)
+  return { types: { $regex: ".*" } }
 }
```
Comment (peetucket) [inline src/endpoints/metrics.js:27]: Same result, but allows us to easily add to the match query in other endpoints, such as adding a filter by group to the query

File: __tests__/endpoints/metrics.test.js [hunk 2]
```diff
@@ -89,3 +92,26 @@ describe("GET /metrics/resourceCount", () => {
...
+    const res = await request(app)
+      .get("/metrics/createdCount/all?startDate=2021-10-01&endDate=2021-11-01")
+      .set("Accept", "application/json")
+    console.log(res)
+    expect(res.statusCode).toEqual(200)
+    expect(res.type).toEqual("application/json")
+    expect(res.body).toEqual(response)
+    expect(mockResponse).toHaveBeenCalledWith(
+      expect.objectContaining({ $match: allResourceQuery })
+    )
```
Comment (peetucket) [inline __tests__/endpoints/metrics.test.js:115]: Need help here, I don't want to add an expectation for the entire big `aggregation` query, was hoping to just verify some subset of it, but can't get the syntax.

File: openapi.yml [hunk 3]
```diff
@@ -74,7 +74,40 @@ paths:
...
+      summary: Get the total number of resources created in a specified time period, optionally filtered by group
+      tags:
+        - metrics
+      responses:
+        "200":
+          description: The number of resources created during a specified time period
+          content:
+            application/json:
+              schema:
+                type: object
```
Comment (justinlittman) [inline openapi.yml:88]: Can you provide a schema for the response?

**Review by justinlittman — CHANGES_REQUESTED** [review]

File: openapi.yml (hunk 3 above)
Comment (peetucket) [reply openapi.yml]: Yes, this was an oversight, thanks.  I added a shared schema for the metrics endpoints (all of which have the same response at the moment) and referenced here.

**Review by justinlittman — APPROVED** [review]

**Pull request merged**